packages/typespec-client-generator-core/src/internal-utils.ts

```diff
@@ -591,3 +592,27 @@ export function hasNoneVisibility(context: TCGCContext, type: ModelProperty): bo
   const visibility = getVisibilityForClass(context.program, type, lifecycle);
   return visibility.size === 0;
 }
+
+export function getAllUserDefinedNamespaces(
+  context: TCGCContext,
+  namespace?: Namespace,
+  retval?: Namespace[],
+): Namespace[] {
+  if (!retval) {
```

Comment:
Is there any way we can use this typekit to filter non user defined namespaces?

https://github.com/microsoft/typespec/blob/main/packages%2Fcompiler%2Fsrc%2Fexperimental%2Ftypekit%2Fkits%2Ftype.ts#L127

Reply:
ooh that's a great idea, let me try that and see what happens